### PR TITLE
Step 2: Add very basic CRUD API for Todos (file-system approach)

### PR DESCRIPTION
**Pull request by vladislav-shkutovich**
The same API:
Create: `/todos` **POST**
Read: `/todos` **GET**, `/todos/:id` **GET**
Update: `/todos/:id` **PATCH**
Delete: `/todos/:id` **DELETE**

Body example:
```
{
    "id": "new/updated id",
    "title": "new/updated todo title",
    "description": "new/updated todo description"
}
```

**What's new**: now all data stored into `./todos.json` file. Reading triggered by the `onModuleInit`, writing triggered by the `onModuleDestroy`.

### CODE REVIEW

File: src/todos/todos.database.service.ts [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
-import { initialTodos } from './mocks'
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+  OnModuleInit,
+  OnModuleDestroy,
+} from '@nestjs/common'
+import * as path from 'path'
+import * as fs from 'fs/promises'
```
Comment (vladislav-shkutovich) [inline src/todos/todos.database.service.ts]: TBD if it's a correct way to import these packages

File: src/todos/todos.database.service.ts [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
 @Injectable()
-export class TodosDatabaseService {
-  static todos: Todos = initialTodos
+export class TodosDatabaseService implements OnModuleInit, OnModuleDestroy {
+  private static todos: Todos = new Map()
+
+  private readonly todosFilePath: string = path.resolve(
+    __dirname,
+    '../../../todos.json',
+  )
```
Comment (vladislav-shkutovich) [inline src/todos/todos.database.service.ts]: TBD if it's a correct way to resolve pathes

File: src/todos/todos.database.service.ts (hunk 1 above)
Comment (masontikhonov) [reply src/todos/todos.database.service.ts]: Nowadays it's preferable to prefix Node's modules with `node:` prefix: `import * as fs from 'node:fs/promises'`
Also, it's always better to import only that parts that you need: `import { readFile, writeFile } from 'node:fs/promises'` — unless you need most of the exported entities.

File: src/todos/todos.database.service.ts (hunk 2 above)
Comment (masontikhonov) [reply src/todos/todos.database.service.ts]: It's correct; one of the approaches.

**Review by masontikhonov — APPROVED** [review]

File: src/todos/todos.database.service.ts (hunk 1 above)
Comment (vladislav-shkutovich) [reply src/todos/todos.database.service.ts]: Thanks! I updated these imports as you advised

**Pull request merged**